### ISSUE TITLE
Add support to azurerm_monitor_diagnostic_setting for resource specific destination table option

### ISSUE THREAD

**Issue by lukerosser** [terraform-providers/terraform-provider-azurerm#4304]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Writing diagnostic logs to log analytics is moving away from a shared diagnostic table and adding resource specific tables. Add a parameter to the diagnostics settings resource to choose where to write log analytics diagnostics logs.

References:

https://azure.microsoft.com/en-gb/updates/adf-logs-in-dedicated-tables/
https://docs.microsoft.com/en-us/azure/azure-monitor/platform/diagnostic-logs-stream-log-store#azure-diagnostics-vs-resource-specific

### New or Affected Resource(s)

* azurerm_monitor_diagnostic_setting

### Potential Terraform Configuration

```hcl
resource "azurerm_monitor_diagnostic_setting" "datafactory_diagnostics" {
  name                           = "resource-specific-diagnostics-table"
  target_resource_id             = azurerm_data_factory.datafactory.id
  log_analytics_workspace_id     = azurerm_log_analytics_workspace.workspace.id
  log_analytics_destination_type = "Resource-Specific"

  log {
    category = "ActivityRuns"

    retention_policy {
      enabled = false
    }
  }

  metric {
    category = "AllMetrics"

    retention_policy {
      enabled = false
    }
  }
}
```

**Comment by KimShepherdVibrato** [comment]
This functionality was released in v1.33.0.

https://github.com/terraform-providers/terraform-provider-azurerm/pull/3987

**Comment by tombuildsstuff** [comment]
@KimShepherdVibrato thanks for the heads-up 👍
